src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = ({ ...state });
+  const renovatedState = [];
+
+  for (const action of actions) {
+    const extraDatavalue = action.extraData || {};
+    const typeOfAction = action.type;
+    const removeValue = action.keysToRemove || {};
+
+    switch (typeOfAction) {
+      case 'addProperties':
+        for (const newProperty in extraDatavalue) {
+          newState[newProperty] = extraDatavalue[newProperty];
+        }
+
+        break;
+
+      case 'removeProperties':
+        for (let j = 0; j < removeValue.length; j++) {
```

Comment:
let's use [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) for this case